File: module-server/src/main/java/de/twomartens/wahlrecht/mapper/v1/ElectedMapper.java
package de.twomartens.wahlrecht.mapper.v1;

import de.twomartens.wahlrecht.model.internal.Elected;
import org.mapstruct.CollectionMappingStrategy;
import org.mapstruct.Mapper;
import org.mapstruct.NullValueCheckStrategy;
import org.mapstruct.ReportingPolicy;

@Mapper(collectionMappingStrategy = CollectionMappingStrategy.ADDER_PREFERRED,
    nullValueCheckStrategy = NullValueCheckStrategy.ALWAYS,
    unmappedTargetPolicy = ReportingPolicy.IGNORE)
public interface ElectedMapper {

  de.twomartens.wahlrecht.model.dto.v1.Elected mapToExternal(Elected elected);

  Elected mapToInternal(de.twomartens.wahlrecht.model.dto.v1.Elected elected);
}

File: module-server/src/main/kotlin/de/twomartens/wahlrecht/mapper/v1/ElectedMapper.kt
package de.twomartens.wahlrecht.mapper.v1

import de.twomartens.wahlrecht.model.internal.Elected
import org.mapstruct.CollectionMappingStrategy
import org.mapstruct.Mapper
import org.mapstruct.NullValueCheckStrategy
import org.mapstruct.ReportingPolicy

@Mapper(
    collectionMappingStrategy = CollectionMappingStrategy.ADDER_PREFERRED,
    nullValueCheckStrategy = NullValueCheckStrategy.ALWAYS,
    unmappedTargetPolicy = ReportingPolicy.IGNORE
)
interface ElectedMapper {
    fun mapToExternal(elected: Elected): de.twomartens.wahlrecht.model.dto.v1.Elected

    fun mapToInternal(elected: de.twomartens.wahlrecht.model.dto.v1.Elected): Elected
}